merlin/study/celeryadapter.py

```diff
@@ -178,10 +178,84 @@ def get_workers_from_app():
     return [*workers]
 
 
-def start_celery_workers(spec, steps, celery_args, just_return_command):
+def _get_workers_to_start(spec, steps):
+    """
+    Helper function to return a set of workers to start based on
+    the steps provided by the user.
+
+    :param `spec`: A MerlinSpec object
+    :param `steps`: A list of steps to start workers for
+
+    :returns: A set of workers to start
+    """
+    workers_to_start = []
+    step_worker_map = spec.get_step_worker_map()
+    for step in steps:
+        try:
+            workers_to_start.extend(step_worker_map[step])
+        except KeyError:
+            LOG.warning(f"Cannot start workers for step: {step}. This step was not found.")
+
+    workers_to_start = set(workers_to_start)
+    LOG.debug(f"workers_to_start: {workers_to_start}")
+
+    return workers_to_start
+
+
+def _create_kwargs(spec):
+    """
+    Helper function to handle creating the kwargs dict that
+    we'll pass to subprocess.Popen when we launch the worker.
+
+    :param `spec`: A MerlinSpec object
+    :returns: A tuple where the first entry is the kwargs and
+              the second entry is variables defined in the spec
+    """
+    senv = spec.environment
+    spenv = os.environ.copy()
```

Comment:
Could we use more descriptive names for senv, spenv, and yenv?
(and yes I know these are legacy variables cut from elsewhere, but should use this opportunity to clean them up a bit)